### PR TITLE
BUG: Wrap new enum classes

### PR DESCRIPTION
**Pull request by thewtex**

### CODE REVIEW

**Comment by thewtex** [comment]
@hjmjohnson since it has an impact on the Python API, and to make improvements for the ITK API, I am considering making the name of the strongly typed enum's consistent:

`<Description>Enum`

The Python API for strongly typed enum's is

`<Description>Enum_<EnumValue1>`
`<Description>Enum_<EnumValue2>`
[...]

That is, `Enum` comes at the end, consistently. We remove `Type` that is at the end currently for brievity (everything is strongly typed). What do you think?

**Comment by hjmjohnson** [comment]
> @hjmjohnson since it has an impact on the Python API, and to make improvements for the ITK API, I am considering making the name of the strongly typed enum's consistent:
> 
> `<Description>Enum`
> 
> The Python API for strongly typed enum's is
> 
> `<Description>Enum_<EnumValue1>`
> `<Description>Enum_<EnumValue2>`
> [...]
> 
> That is, `Enum` comes at the end, consistently. We remove `Type` that is at the end currently for brievity (everything is strongly typed). What do you think?

GREAT IDEA!  Doing this soon would not be overly disruptive since the strongly typed enums have not been around that long.

File: Wrapping/Generators/Python/itkExtras.py [hunk 1]
```diff
@@ -552,7 +552,7 @@ def imread(filename, pixel_type=None, fallback_only=False):
         increase_dimension=False
         kwargs={'FileName':filename}
     if pixel_type:
-        imageIO = itk.ImageIOFactory.CreateImageIO(io_filename, itk.ImageIOFactory.ReadMode)
+        imageIO = itk.ImageIOFactory.CreateImageIO(io_filename, itk.ImageIOFactory.FileModeType_ReadMode)
```
Comment (hjmjohnson) [inline Wrapping/Generators/Python/itkExtras.py]: This is a good change I think.  It greatly helps reduce name conflicts.

File: Wrapping/Generators/SwigInterface/igenerator.py [hunk 2]
```diff
@@ -585,13 +582,13 @@ def generate_enum(self, typedef):
...
-        self.outputFile.write("enum %s { %s };\n" % (name, ", ".join(content)))
+        content = [" %s" % (key,) for key, value in enum.values]
+        self.outputFile.write("enum class %s: uint8_t { %s };\n\n" % (name, ", ".join(content)))
 
     def generate_nested_enum(self, typedef, enum, indent, w):
-        content = [" %s = %i" % (key, value) for key, value in enum.values]
+        content = [" %s" % (key,) for key, value in enum.values]
         self.outputFile.write("  " * indent)
-        self.outputFile.write("    enum %s { %s };\n" % (enum.name, ", ".join(content)))
+        self.outputFile.write("    enum class %s: uint8_t { %s };\n\n" % (enum.name, ", ".join(content)))
```
Comment (hjmjohnson) [inline Wrapping/Generators/SwigInterface/igenerator.py]: AHHH!  This is the magic fix that I could not track down!  I could not figure out where I was going wrong. 

I hope this solution works.

**Comment by thewtex** [comment]
> > @hjmjohnson since it has an impact on the Python API, and to make improvements for the ITK API, I am considering making the name of the strongly typed enum's consistent:
> > `<Description>Enum`
> > The Python API for strongly typed enum's is
> > `<Description>Enum_<EnumValue1>`
> > `<Description>Enum_<EnumValue2>`
> > [...]
> > That is, `Enum` comes at the end, consistently. We remove `Type` that is at the end currently for brievity (everything is strongly typed). What do you think?
> 
> GREAT IDEA! Doing this soon would not be overly disruptive since the strongly typed enums have not been around that long.

Cool -- I will add this in another PR, then! We are making great strides.

File: Wrapping/Generators/SwigInterface/igenerator.py (hunk 2 above)
Comment (thewtex) [reply Wrapping/Generators/SwigInterface/igenerator.py]: It is great to have more folks understanding this process. I did not know how enum's worked in the wrapping process previously, but now appears to be quite simple.

In `igenerator.py`, we create the SWIG interface *.i* files that define the Python interface. We have a CastXML data structure that represents the abstract syntax tree. We take the enum's and tell SWIG to generate the interface for strongly typed enum's.

**Pull request merged**